replica/table.cc

```diff
@@ -954,7 +971,7 @@ table::update_sstable_lists_on_off_strategy_completion(sstables::compaction_comp
 
 // Note: must run in a seastar thread
 void
-table::on_compaction_completion(sstables::compaction_completion_desc desc) {
+table::on_compaction_completion(compaction_group& cg, sstables::compaction_completion_desc desc) {
```

Comment:
My gut feeling is that this method should move to compaction_group
and that should interact with the table layer to execute the table-wide code.

Reply:
Shouldn't it be table coordinating the activity rather than the other way around? 

the update of sstable set happens through row_cache::invalidate(external_updater, ...) method, and we have table then refreshing its cache snapshot, updating its statistics, submitting files for deletion, etc.

Maybe sstable_list_updater (which implements the actual replacement), which is defined in table::on_compaction_completion() should be moved to compaction_group. Maybe it's not needed.

Reply:
I think that compaction_manager should deal with compaction_group_state that should interact with the compaction_group object, and if that needs interaction with the table, so be it.
But bypassing that and calling the table with a refwrence to the compaction group seems like violating layering.